Doc/library/re.rst

```diff
@@ -496,6 +496,8 @@ The special characters are:
 
    .. versionchanged:: 3.12
       Group *id* can only contain ASCII digits.
+      In bytes replacement strings group names must contain only characters
+      in the ASCII range.
```

Comment:
Maybe its just me, but this was pretty confusing to me and hard to parse and understand. The comma @merwok suggest helps a lot in making the basic grammar more clear, but there still seems to be some problems with the actual semantics here.

As far as I can tell, this is referring to _patterns_, not replacement strings, and I was very confused why the latter were mentioned until I realized it was incorrectly copy/pasted from the original mention (that talks about replacement strings).

Also, "bytes strings" is both arguably an oxymoron in Python 3 (since this is presumably intended to specifically refer to bytes, not strings), seemingly ungrammatical (at least without being proper type references), and certainly confusing.

Finally, *name* should be formatted consistently with *id* in the previous line, :class:`bytes` objects are composed of bytes, not characters, and it would be useful to explictly specify what the ASCII range is. Implementing all of that, and fixing the phrasing, we have:

```suggestion
      In :class:`bytes` patterns, group *name* can only contain bytes
      in the ASCII range (`\x00`-`\x7f`).
```

Reply:
@CAM-Gerlach Maybe `b'\x00'`-`b'\x7f'`?

Reply:
I guess so; edited. I just didn't want to overcomplicate it.